.github/workflows/codeql-analysis.yml

```diff
@@ -29,15 +29,15 @@ jobs:
 
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
-      uses: github/codeql-action/init@v1
+      uses: github/codeql-action/init@v3
```

Comment:
With the CodeQL actions, on the other hand, I've found it more difficult to determine the correct SHA for the version I want. So I'm just going with `v3` or similar for these. (I don't view the SHA as mandatory; just a nice-to-have.)